lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -67,11 +67,18 @@ class WindowsFirewallRule < Chef::Resource
         description: "The action of the firewall rule.",
         coerce: proc { |f| f.is_a?(String) ? f.downcase.to_sym : f }
 
-      property :profile, [Symbol, String],
-        default: :any, equal_to: %i{public private domain any notapplicable},
+      property :profile, [Symbol, String, Array],
+        default: :any,
         description: "The profile the firewall rule applies to.",
-        coerce: proc { |p| p.is_a?(String) ? p.downcase.to_sym : p }
-
+        coerce: proc { |p|
+          if p.is_a?(Array)
+            p.map(&:downcase).map(&:to_sym).sort
+          elsif p.is_a?(String)
+            Array(p.downcase.to_sym).sort
+          else
+            Array(p).sort
+          end
```

Comment:
`coerce: proc { |p| Array(p).map(&:downcase).map(&:to_sym).sort }` will probably do this without the conditional.

Reply:
Yeah, that's much better